packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -62,13 +65,48 @@ async function checkProofArgs(args: GPCPCDArgs): Promise<{
   }
   const proofConfig = deserializeGPCProofConfig(args.proofConfig.value);
 
-  if (!args.pod.value) {
+  // Filter out PODPCD records.
```

Comment:
This is tricky and clever.
Maybe too clever, since it took me a while to puzzle over what it's doing.
What do you think of an alternative like this (may not compile, but you can hopefully get the gist:
```ts
    const podRecord = Object.fromEntries(Object.entries(args).filter(([n, _v]) => n.startsWith(PODPCD_ARG_PREFIX)));
```

Reply:
That is reasonable, though I was thinking it would be good to catch invalid arguments and throw as I do further down, but perhaps that is unnecessary. I'll make the change.

Reply:
Ah, fair point.  Catching invalid arguments is a reasonable thing to do, though mixing it with extracting the podpcd arguments seems a bit odd.  In the case of PCD requests we'll certainly notice missing args, but not unrecognized ones.  The nature of how TypeScript types work does tend to encourage ignoring unknown fields rather than being strict about them.  I didn't put any unknown field checking in the GPC types (like GPCProofConfig) though I wonder whether I should at some point.  It depends how we approach compatibility, I suppose.  I.e. if we add a new field, do we want clients which don't know about it to reject the config or treat it like an old version?